api/valor_api/backend/core/llm_clients.py

```diff
@@ -0,0 +1,659 @@
+**
+
+Text:
+{text}
+
+JSON:
+"""
+
+
+def _generate_anwswer_relevance_verdicts_instruction(
+    query: str, statements: str | list[str]
+) -> str:
+    """
+    Instruction template was copied from DeepEval's codebase https://github.com/confident-ai/deepeval/blob/main/deepeval/metrics/answer_relevancy/template.py.
+
+    Parameters
+    ----------
+    query: str
+        The query to evaluate the statements against.
+    statements: str
+        The statements to evaluate the validity of.
+
+    Returns
+    -------
+    str
+        The instruction for the llm.
+    """
+    return f"""For the provided list of statements, determine whether each statement is relevant to address the input.
+Please generate a list of JSON with two keys: `verdict` and `reason`.
+The 'verdict' key should STRICTLY be either a 'yes', 'idk' or 'no'. Answer 'yes' if the statement is relevant to addressing the original input, 'no' if the statement is irrelevant, and 'idk' if it is ambiguous (eg., not directly relevant but could be used as a supporting point to address the input).
+The 'reason' is the reason for the verdict.
+Provide a 'reason' ONLY if the answer is 'no'.
+The provided statements are statements made in the actual output.
+
+**
+IMPORTANT: Please make sure to only return in JSON format, with the 'verdicts' key mapping to a list of JSON objects.
+Example input: What should I do if there is an earthquake?
+Example statements: ["Shoes.", "Thanks for asking the question!", "Is there anything else I can help you with?", "Duck and hide"]
+Example JSON:
+{{
+    "verdicts": [
+        {{
+            "verdict": "no",
+            "reason": "The 'Shoes.' statement made in the actual output is completely irrelevant to the input, which asks about what to do in the event of an earthquake."
+        }},
+        {{
+            "verdict": "idk"
+        }},
+        {{
+            "verdict": "idk"
+        }},
+        {{
+            "verdict": "yes"
+        }}
+    ]
+}}
+
+Since you are going to generate a verdict for each statement, the number of 'verdicts' SHOULD BE STRICTLY EQUAL to the number of `statements`.
+**
+
+Input:
+{query}
+
+Statements:
+{statements}
+
+JSON:
+"""
+
+
+def _get_coherence_instruction(text: str) -> str:
+    """
+    This instruction was adapted from appendix A of DeepEval's paper G-EVAL: NLG Evaluation using GPT-4 with Better Human Alignment (https://arxiv.org/pdf/2303.16634).
+    The main adaptation is a generalization of the metric to more task types. The example prompt in DeepEval was specific to summarization, but the below prompt could apply to any text generation task.
+    Crucially, unlike DeepEval, no context is used. Instead, the coherence of the text is evaluated entirely based on the text. This generalizes the prompt and also prevents the evaluation from being influenced by the quality of sentences in the context.
+    Parameters
```

Comment:
nit: add a newline here between the text and "Parameters"? can't tell if one exists or not